### DISCUSSION TITLE
Ext Opts Jump Index

### DISCUSSION THREAD

**Discussion by austinliuigi** [L3MON4D3/LuaSnip#916] [Q&A] [answered]
I would like to use inline extmarks to show the jump locations and order of nodes in my snippets. Is it possible to:

1. Get the column number of the beginning of the node so that I know where to place the extmark
2. Set the text to be the jump index for all nodes without overriding the ext_opts for each node?

Thanks

**Comment by L3MON4D3** [comment, marked as answer]
(1.) should not be necessary, see below (at least for inserting the inlay-hint for all insertNodes)
(2.) is not really possible, unfortunately

But this can be done somewhat comfortably if you're using the lua-loader: override the default `i` (in `snip_env`) with a function which also sets `ext_opts` on the insertNode before returning it.

One issue with this is that this is not global, but only applies to snippets loaded via lua.

It might be worth it to look into generating `ext_opts` more dynamically, or at least provide some global settings for common stuff (like showing the jump-order via inlay hints)

**Reply by austinliuigi** [reply]
Got it, thanks!
